packages/trpc/server/routers/loggedInViewer/updateProfile.handler.ts

```diff
@@ -234,3 +230,21 @@ const handleUserMetadata = ({ ctx, input }: UpdateProfileOptions) => {
   // Required so we don't override and delete saved values
   return { ...userMetadata, ...cleanMetadata };
 };
+
+async function getAvatarToSet(avatar: string | null | undefined) {
+  if (avatar) {
+    if (avatar.startsWith("data:image")) {
+      return await resizeBase64Image(avatar);
+    } else {
+      // Non Base64 avatar currently could only be the dynamic avatar URL(i.e. /{USER}/avatar.png). If we allow setting that URL, we would get infinite redirects on /user/avatar.ts endpoint
+      log.warn("Non Base64 avatar, ignored it", { avatar });
+      // `undefined` would not ignore the avatar, but `null` would remove it. So, we return `undefined` here.
+      return undefined;
+    }
+  }
+
+  if (avatar === null) {
+    return null;
+  }
+  return avatar;
+}
```

Comment:
I think we can rewrite this function to be more clear by returning early and avoiding nesting

```
if avatar is null => return null

if avatar does not start with data:image =>  return undefined

return await resizeBase64Image(avatar);

```

Reply:
Done